system/lib/pthread/pthread_create.c

```diff
@@ -328,6 +328,10 @@ void _emscripten_thread_exit(void* result) {
     // Mark the thread as no longer running, so it can be joined.
     // Once we publish this, any threads that are waiting to join with us can
     // proceed and this worker can be recycled and used on another thread.
+#ifdef __PIC__
+    // When dynamic linking is enabled we need to keep track of zombie threads
+    _emscripten_thread_exit_joinable(self);
+#endif
     a_store(&self->detach_state, DT_EXITED);
```

Comment:
The `self->detach_state` looks like would be the root info to detect an unjoined nondetached thread? So instead of looking for `PThread.zombieThreads`, JS code could do a load in `pthread_ptr->detach_state` and look for `DT_EXITED` there? This would keep a single source of truth?

Reply:
I renamed `zombieThreads` to `finishedThreads` to avoid using the term zombie at all.

I did keep this new `_emscripten_thread_exit_joinable` method though since the main thread needs to some how know asynchronously when a thread enters this state. 

Alternatively the main thread to periodically iterate through all threads to see if any of them have entered this state (perhaps using setTimeout or something?) .   But this asynchronous notification seem more efecient.